python/pyarrow/tests/test_convert_builtin.py

```diff
@@ -2363,3 +2363,136 @@ def test_array_from_pylist_offset_overflow():
     assert isinstance(arr, pa.ChunkedArray)
     assert len(arr) == 2**31
     assert len(arr.chunks) > 1
+
+
```

Comment:
Great test cases! I see we aren't testing all scalar types defined in https://arrow.apache.org/docs/python/api/datatypes.html, but I don't think we need to since many of them take identical code paths, right? e.g. date32 vs date64

Reply:
Yeah, I took approx one example for the group of datatypes represented in the specific `PyPrimitiveConverter` class that holds an `Append` method in `python/pyarrow/src/arrow/python/python_to_arrow.cc`. There should be no difference between for example `date32` vs `date64`. But will add an extra case to the test just to be sure 👍